napari/_vispy/layers/labels.py

```diff
@@ -195,17 +195,17 @@ def _get_shape_from_dict(color_dict):
 def get_shape_from_dict(color_dict):
     global MAX_TEXTURE_SIZE
     if MAX_TEXTURE_SIZE is None:
-        MAX_TEXTURE_SIZE = get_max_texture_sizes()
+        MAX_TEXTURE_SIZE = min(get_max_texture_sizes())
```

Comment:
min? Wha tif one or the other is None? Why not just get the 2d (`[0]`)?

Reply:
for example on my machine 2d is bigger than 3d:

```python 
from napari._vispy.utils.gl import get_max_texture_sizes
print(get_max_texture_sizes())
```

provides: `(32768, 16384)`

Reply:
But we're using a 2D texture, so it doesn't matter what the allowed 3D shape is.

Reply:
I guess you mean `sampler3D`, in which case yes. Regardless whether we look at 3D or 2D labels, the lookup table for coloring is a 2D texture!

Reply:
fixed